src/main/java/com/mjd/bank/services/impl/AccountServiceImpl.java

```diff
@@ -35,4 +45,19 @@ public SimpleMessageResponse deposit(Long ownerId, Long accountId, BigDecimal am
 
     return new SimpleMessageResponse(String.format("Deposit of %s to account %s completed", amount, accountId));
   }
+
+  @Override
+  public SimpleMessageResponse create(Long ownerId, AccountType type) {
+
+    AppUser owner = appUserRepository.findById(ownerId)
+            .orElseThrow(() ->new NotFoundException("The user with ID " + ownerId + " doesn't exist"));
+
+      if (type!=AccountType.UNKNOWN){
+        Account account = new Account(type,owner);
+        accountRepository.save(account);
+        return new SimpleMessageResponse("Account created successfully");
+      }else {
+        throw new IncorrectAccountTypeException("The account type doesn't exist");
+      }
+  }
```

Comment:
Te sugiero invertir el if else para que quede más legible. Y además comprobar que el usuario sí sea el dueño de la cuenta.
```suggestion
  @Override
  public SimpleMessageResponse create(Long ownerId, AccountType type) {

    AppUser owner = appUserRepository.findById(ownerId)
            .orElseThrow(() ->new NotFoundException("The user with ID " + ownerId + " doesn't exist"));
            
      transactionsUtils.isAccountOwner(ownerId, account.getOwner().getId());

      if (type == AccountType.UNKNOWN) {
        throw new IncorrectAccountTypeException("The account type doesn't exist");
      }
      
       Account account = new Account(type, owner);
       accountRepository.save(account);
       return new SimpleMessageResponse("Account created successfully");
  }
```